salt/haproxy/init.sls

```diff
@@ -0,0 +1,48 @@
+---
+/etc/haproxy/haproxy.cfg:
+  file.managed:
+    - source: salt://haproxy/haproxy.cfg.jinja
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+
+haproxy:
+  file.managed:
+    - name: /etc/kubernetes/manifests/haproxy.yaml
+    - source: salt://haproxy/haproxy.manifest
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+{% if "kube-master" in salt['grains.get']('roles', []) %}
+  iptables.append:
+{% else %}
+  iptables.delete:
```

Comment:
Preface - I think you just spotted an issue, I'll need to double check in the morning.

Oh, let me rephrase slightly. We do listen on 6443 on all machines - however on masters, it's 0.0.0.0:6443, and on workers it's 127.0.0.1:6443.

The actual kube-api process listens on 0.0.0.0:6444 on the masters, and is not bound at all on the workers.

The issue I think I have is - I'm only adding 1 allow rule, when I need two (and eventually, we get network segregation, and the second allow rule is on the "internal" network)